lib/seq.ml

```diff
@@ -0,0 +1,51 @@
+(*
+ * Copyright (c) 2014 Richard Mortier <mort@cantab.net>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ *)
+
+type 'a t = unit -> 'a option
+type buf = Cstruct.t
+
+let iter lenfn pfn t =
```

Comment:
Yes, it's catching `Invalid_argument`.
I'd like to upstream better error handling to `Cstruct` at some point, but haven't figured it out yet. It probably involves the ongoing how-we-want-to-write-our-parsers discussions...
(I've since realised that merging `buf.ml` and `seq.ml` probably at least makes sense. Worth doing?)